src/modules/hip/kernel/gaussian_filter.hpp

```diff
@@ -0,0 +1,2235 @@
+    }
+    __syncthreads();
+    if ((id_x_o < roiTensorPtrSrc[id_z].xywhROI.roiWidth) &&
+        (id_y_o < roiTensorPtrSrc[id_z].xywhROI.roiHeight) &&
+        (hipThreadIdx_x < tileSize.x) &&
+        (hipThreadIdx_y < tileSize.y))
+    {
+        gaussian_filter_5x5_row_hip_compute(&srcShared[hipThreadIdx_y_channel.x    ][hipThreadIdx_x8], &sum_f24.f8[0], filter_row1);
+        gaussian_filter_5x5_row_hip_compute(&srcShared[hipThreadIdx_y_channel.y    ][hipThreadIdx_x8], &sum_f24.f8[1], filter_row1);
+        gaussian_filter_5x5_row_hip_compute(&srcShared[hipThreadIdx_y_channel.z    ][hipThreadIdx_x8], &sum_f24.f8[2], filter_row1);
+        gaussian_filter_5x5_row_hip_compute(&srcShared[hipThreadIdx_y_channel.x + 1][hipThreadIdx_x8], &sum_f24.f8[0], filter_row2);
+        gaussian_filter_5x5_row_hip_compute(&srcShared[hipThreadIdx_y_channel.y + 1][hipThreadIdx_x8], &sum_f24.f8[1], filter_row2);
+        gaussian_filter_5x5_row_hip_compute(&srcShared[hipThreadIdx_y_channel.z + 1][hipThreadIdx_x8], &sum_f24.f8[2], filter_row2);
+        gaussian_filter_5x5_row_hip_compute(&srcShared[hipThreadIdx_y_channel.x + 2][hipThreadIdx_x8], &sum_f24.f8[0], filter_row3);
+        gaussian_filter_5x5_row_hip_compute(&srcShared[hipThreadIdx_y_channel.y + 2][hipThreadIdx_x8], &sum_f24.f8[1], filter_row3);
+        gaussian_filter_5x5_row_hip_compute(&srcShared[hipThreadIdx_y_channel.z + 2][hipThreadIdx_x8], &sum_f24.f8[2], filter_row3);
+        gaussian_filter_5x5_row_hip_compute(&srcShared[hipThreadIdx_y_channel.x + 3][hipThreadIdx_x8], &sum_f24.f8[0], filter_row4);
+        gaussian_filter_5x5_row_hip_compute(&srcShared[hipThreadIdx_y_channel.y + 3][hipThreadIdx_x8], &sum_f24.f8[1], filter_row4);
+        gaussian_filter_5x5_row_hip_compute(&srcShared[hipThreadIdx_y_channel.z + 3][hipThreadIdx_x8], &sum_f24.f8[2], filter_row4);
+        gaussian_filter_5x5_row_hip_compute(&srcShared[hipThreadIdx_y_channel.x + 4][hipThreadIdx_x8], &sum_f24.f8[0], filter_row5);
+        gaussian_filter_5x5_row_hip_compute(&srcShared[hipThreadIdx_y_channel.y + 4][hipThreadIdx_x8], &sum_f24.f8[1], filter_row5);
+        gaussian_filter_5x5_row_hip_compute(&srcShared[hipThreadIdx_y_channel.z + 4][hipThreadIdx_x8], &sum_f24.f8[2], filter_row5);
+        rpp_hip_adjust_range(dstPtr, &sum_f24);
+        rpp_hip_pack_float24_pln3_and_store24_pkd3(dstPtr + dstIdx, &sum_f24);
+    }
+}
+
+// kernelSize = 7
+template <typename T>
+__global__ void gaussian_filter_7x7_pkd_tensor(T *srcPtr,
+                                               uint2 srcStridesNH,
+                                               T *dstPtr,
+                                               uint2 dstStridesNH,
+                                               uint padLength,
+                                               uint2 tileSize,
+                                               RpptROIPtr roiTensorPtrSrc,
+                                               d_float49 *filterTensor)
```

Comment:
I would suggest not using this type for function argument. Instead take it as float *

Reply:
Done